build.gradle

```diff
@@ -40,41 +26,60 @@ logger.lifecycle("Building Energy: " + version)
 
 repositories {
 	mavenCentral()
+	maven {
+		name = "Technici4n"
+		url = "https://raw.githubusercontent.com/Technici4n/Technici4n-maven/master/"
+		content {
+			includeGroup "net.fabricmc.fabric-api"
+		}
+	}
 }
 
 dependencies {
-	minecraft "com.mojang:minecraft:1.16.4"
-	mappings "net.fabricmc:yarn:1.16.4+build.7:v2"
+	minecraft "com.mojang:minecraft:${project.minecraft_version}"
+	mappings "net.fabricmc:yarn:${project.yarn_version}:v2"
 
 	modImplementation  "net.fabricmc:fabric-loader:0.10.8"
 
-	testCompile 'junit:junit:4.12'
-}
+	// Temporary WIP fabric api fork
+	include modApi("net.fabricmc.fabric-api:fabric-api-lookup-api-v1:1.3.0+b246072f13")
+	include modApi("net.fabricmc.fabric-api:fabric-transfer-api-v1:1.1.0-pre.06+c35ca56f18")
+	modApi "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 
-processResources {
-	inputs.property "version", project.version.toString()
+	testCompileOnly 'junit:junit:4.12'
+}
 
-	from(sourceSets.main.resources.srcDirs) {
-		include "fabric.mod.json"
-		expand "version": project.version.toString()
+sourceSets {
+	test {
+		compileClasspath += sourceSets.main.output
+		runtimeClasspath += sourceSets.main.output
 	}
+	jmh {
+		compileClasspath += sourceSets.test.output
+		runtimeClasspath += sourceSets.test.output
+	}
+}
+
+processResources {
+	inputs.property "version", project.version
 
-	from(sourceSets.main.resources.srcDirs) {
-		exclude "fabric.mod.json"
+	filesMatching("fabric.mod.json") {
+		expand "version": project.version
 	}
 }
 
-tasks.withType(JavaCompile) {
-	options.encoding = "UTF-8"
+tasks.withType(JavaCompile).configureEach {
+	it.options.encoding = "UTF-8"
+	it.options.release =16
```

Comment:
If we are doing this make sure to set the min version in the fabric.mod.json file, along with mc 1.17

Reply:
Done.